src/ContainerCommitModal.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import {Modal, Button} from 'react-bootstrap';
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+class ContainerCommitModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.initialState = {
+            imageName: "",
+            tag: "",
+            author:"",
+            message: "",
+            command: this.props.containerWillCommit.Config ? this.props.containerWillCommit.Config.Cmd.join(" ") : "",
+            pause: true,
+            setonbuild: false,
+            onbuild: [""],
+            format: "oci",
+            selectedFormat: "oci",
+            onbuildDisabled: true,
+        };
+        this.state = {...this.initialState};
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleCancel = this.handleCancel.bind(this);
+        this.handleCommit = this.handleCommit.bind(this);
+        this.handleOnBuildsInputChange = this.handleOnBuildsInputChange.bind(this);
+        this.handleAddOnBuild = this.handleAddOnBuild.bind(this);
+        this.handleRemoveOnBuild = this.handleRemoveOnBuild.bind(this);
+    }
+
+    // TODO
+    handleOnBuildsInputChange(idx, evt) {
+
+    }
+
+    handleAddOnBuild() {
+        this.setState({onbuild: this.state.onbuild.concat([""])});
+    }
+
+    handleRemoveOnBuild(idx) {
+        this.setState({onbuild: this.state.onbuild.filter((bud, sidx) => idx !== sidx)});
+    }
+
+    // TODO
+    handleInputChange(event) {
+
```

Comment:
I think you 'll need here to pass also some key to identify which one from the input forms had changed since you 're using this function as callback in multiple places.
But since it's not implemented you can do it when you implement it.